bazel/foreign_cc/vpp_vcl.patch

```diff
@@ -16,16 +17,18 @@
  string(REPLACE "-" ";" VPP_LIB_VERSION ${VPP_VERSION})
  list(GET VPP_LIB_VERSION 0 VPP_LIB_VERSION)
  
-@@ -188,7 +184,7 @@ elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
+@@ -188,8 +184,7 @@ elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
    find_package(OpenSSL)
    set(SUBDIRS
      vppinfra svm vlib vlibmemory vlibapi vnet vpp vat vat2 vcl vpp-api
 -    plugins tools/vppapigen tools/g2 tools/perftool cmake pkg
-+    tools/vppapigen tools/g2 tools/perftool cmake pkg
-     tools/appimage
+-    tools/appimage
++    tools/vppapigen cmake pkg
```

Comment:
im guessing a lot of this is not required, but i guess also that with the right flags/target it would be ignored

Reply:
Components in vpp are unfortunately not as cleanly separated as we'd want. Still, until now we could do with a minimal build because of this snippet in `vpp_vcl_build.sh`:

`cmake -G Ninja ../src -DCMAKE_BUILD_TYPE:STRING=release
ninja -C . vppcom`

In other words, we're only building vppcom lib target and its dependencies, not the whole of vpp. The heavy part is vnet not the binary api. Did the `build_args` in `envoy_cmake` lower not work?

Reply:
not sure what should/not build tbh

it was my first real experience of cmake and i now understand why its so disliked

i tried so many different ways, but cmake's use of it was patchy at best and a long way from the scant docs i could find

ideally we update the cmake config in such a way as we can cleanly call just the target we actually need

if its difficult to do there, its an order of manitude more difficult doing it with patches etc